airbyte-webapp/src/components/ui/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,163 @@
+    return undefined;
+  }
+
+  const date = dayjs.utc(input);
+
+  if (!date?.isValid()) {
+    return undefined;
+  }
+
+  // Get the user's UTC offset based on the local time
+  const browserUtcOffset = dayjs().utcOffset();
+
+  // Convert the selected date into a string which we can use to initialize a new date object.
+  // The second parameter to utcOffset() keeps the same local time, only changing the timezone.
+  const dateInUtcAsString = date.utcOffset(browserUtcOffset, true).format();
+
+  const equivalentDate = dayjs(dateInUtcAsString);
+
+  // dayjs does not 0-pad years when formatting, so it's possible to have an invalid date here
+  // https://github.com/iamkun/dayjs/issues/1745
+  if (!equivalentDate.isValid()) {
+    return undefined;
+  }
+
+  return equivalentDate.toDate();
+};
+
+export interface DatePickerProps {
+  error?: boolean;
+  value: string;
+  onChange: (value: string) => void;
+  withTime?: boolean;
+  disabled?: boolean;
+  onBlur?: (ev: React.FocusEvent<HTMLDivElement>) => void;
+  placeholder?: string;
+}
+
+interface DatePickerButtonTriggerProps {
+  onClick?: () => void;
+}
+
+const DatepickerButton = React.forwardRef<HTMLButtonElement, DatePickerButtonTriggerProps>(({ onClick }, ref) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <Button
+      className={styles.datepickerButton}
+      aria-label={formatMessage({ id: "form.openDatepicker" })}
+      onClick={onClick}
+      ref={ref}
+      type="button"
+      variant="clear"
+      icon={<FontAwesomeIcon icon={faCalendarAlt} className={styles.dropdownButton} fixedWidth />}
+    />
+  );
+});
+
+export const DatePicker: React.FC<DatePickerProps> = ({
+  disabled,
+  error,
+  onChange,
+  onBlur,
+  placeholder,
+  value = "",
+  withTime = false,
+}) => {
+  const { locale } = useIntl();
+  const datepickerRef = useRef<ReactDatePicker>(null);
+
+  // Additional locales can be registered here as necessary
+  useEffect(() => {
+    switch (locale) {
+      case "en":
+        registerLocale(locale, en);
+        break;
+    }
+  }, [locale]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleDatepickerChange = useCallback(
+    (val: Date | null) => {
+      const date = dayjs(val);
+      if (!date.isValid()) {
+        onChange("");
+        return;
+      }
+
+      const formattedDate = withTime ? date.utcOffset(0, true).format() : date.format("YYYY-MM-DD");
+      onChange(formattedDate);
+      inputRef.current?.focus();
+    },
+    [onChange, withTime]
+  );
+
+  const handleInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      onChange(e.target.value);
+    },
+    [onChange]
+  );
+
+  const localDate = useMemo(() => toEquivalentLocalTime(value), [value]);
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const handleWrapperBlur = (event: React.FocusEvent<HTMLDivElement>) => {
+    if (onBlur && !wrapperRef.current?.matches(":focus-within")) {
+      onBlur(event);
```

Comment:
Small note: the `event` here, will usually be the bubbled up one, so not actually of the `HTMLDivElement` (i.e. the `event.target` will be the actual element that lost focus, not the wrapper `div`). Because that could mean we're passing random elements' blur events around here, I feel we should instead just not pass the `event` forward at all. Instead I'd just recommend we `stopPropagation` here and just call `onBlur` without any event.

Reply:
@josephkmh how do you feel about this comment from Tim? I'll hold off on approval and locally testing once more until this is addressed

Reply:
@timroes thanks! That is indeed the wrong generic for the event we're expecting, and you're right, we don't really need to pass the event at all. I've just removed that.

I don't think stopping blur events from bubbling is the best choice here, though. What benefit would that bring? Seems to me it's better to leave the default browser behavior there.